src/appshell/qml/Preferences/PercussionPreferencesPage.qml

```diff
@@ -0,0 +1,161 @@
+                // TODO: configuration - toggle autoShowPercussionPanel
+            }
+        }
+
+        StyledTextLabel {
+            id: padSwapInfo
+
+            width: parent.width
+
+            horizontalAlignment: Text.AlignLeft
+            wrapMode: Text.Wrap
+            text: qsTrc("notation", "When swapping the positions of two drum pads:")
+        }
+
+        RadioButtonGroup {
+            id: radioButtons
+
+            property int navigationRowStart: unpitchedSelectedCheckbox.navigation.row + 1
+            property int navigationRowEnd: radioButtons.navigationRowStart + model.length
+
+            width: parent.width
+            spacing: percussionPanelPreferences.spacing
+
+            orientation: ListView.Vertical
+
+            model: [
+                { text: qsTrc("notation", "Move MIDI notes and keyboard shortcuts with their sounds"), value: true },
+                { text: qsTrc("notation", "Leave MIDI notes and keyboard shortcuts fixed to original pad positions"), value: false }
+            ]
+
+            delegate: Row {
+                width: parent.width
+                spacing: 6
+
+                RoundedRadioButton {
+                    id: radioButton
+
+                    anchors.verticalCenter: parent.verticalCenter
+
+                    navigation.name: modelData.text
+                    navigation.panel: percussionPanelPreferences.navigation
+                    navigation.row: radioButtons.navigationRowStart + model.index
+
+                    // TODO: checked: modelData.value === pad swap preference
+
+                    onToggled: {
+                        // TODO: configuration - change pad swap preference
+                    }
+                }
+
+                //! NOTE: Can't use radioButton.text because it won't wrap
+                StyledTextLabel {
+                    width: parent.width - parent.spacing - radioButton.width
+
+                    anchors.verticalCenter: parent.verticalCenter
+
+                    horizontalAlignment: Text.AlignLeft
+                    wrapMode: Text.Wrap
+                    text: modelData.text
+
+                    MouseArea {
+                        id: mouseArea
+
+                        anchors.fill: parent
+
+                        onClicked: {
+                            // TODO: configuration - change pad swap preference
+                        }
+                    }
+                }
+            }
+        }
+
+        CheckBox {
+            id: alwaysAsk
+
+            width: parent.width
+
+            text: qsTrc("notation", "Always ask")
+
+            navigation.name: "AlwaysAskCheckBox"
+            navigation.panel: percussionPanelPreferences.navigation
+            navigation.row: radioButtons.navigationRowEnd
+
+            onClicked:  {
+                // TODO: configuration - toggle "show pad swap dialog"
+            }
+        }
+    }
+
+    FlatButton {
+        id: useNewPercussionPanel
```

Comment:
I don't really like the fact that we will have to support both the old and new percussion panels. We should get rid of the old panel as soon as possible